src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ExportClientResponse.cs

```diff
@@ -14,6 +14,7 @@ protected ExportClientResponse(bool success, DateTime deadlineUtc, Exception? ex
         this.DeadlineUtc = deadlineUtc;
     }
 
+    [MemberNotNullWhen(false, nameof(Exception))]
```

Comment:
This is interesting. A `bool` can never be `null`. Was there some warning that necessitated this?

Reply:
This mean that Exception property will be not null when this property is false.
Yes, there were warning in couple of places.

Reply:
Ah got it this is about `Exception` property not `Success`. Makes sense.